### PR TITLE
DIA-3249 handle ccpa opt-outs

### PR DESCRIPTION
**Pull request by andresilveirah**
* fixed an issue with `/pv-data` being called for campaigns that are not in the config ✅ 
* fixed an issue with `/consent-status` being called in fresh installs (when `localVersion` is `nil`) ✅ 
* added `transitionCCPAAuth` flag to `SPConfig` so clients can choose to transfer their CCPA authenticated consent to USNat ✅
* implemented logic to transfer CCPA opted-out consent over to USNat ✅

### CODE REVIEW

File: ConsentViewController/Classes/SourcePointClient/ConsentStatusMetadata.swift [hunk 1]
```diff
@@ -13,9 +13,48 @@ struct ConsentStatusMetaData: QueryParamEncodable {
         let applies: Bool
         let dateCreated: SPDate?
         let uuid: String?
-        let hasLocalData: Bool
+        let hasLocalData = false
```
Comment (carmelo-iriti) [inline ConsentViewController/Classes/SourcePointClient/ConsentStatusMetadata.swift:16]: do we need to set a default?

File: ConsentViewController/Classes/SourcePointClient/SourcepointClientCoordinator.swift [hunk 2]
```diff
@@ -698,21 +716,21 @@ class SourcepointClientCoordinator: SPClientCoordinator {
 
     func pvData(pubData: SPPublisherData?, _ handler: @escaping () -> Void) {
         let pvDataGroup = DispatchGroup()
-        if let gdprMetadata = state.gdprMetaData {
+        if let gdprMetadata = state.gdprMetaData, campaigns.gdpr != nil {
```
Comment (carmelo-iriti) [inline ConsentViewController/Classes/SourcePointClient/SourcepointClientCoordinator.swift:719]: Is this to filter campaign?

File: ConsentViewController/Classes/SourcePointClient/ConsentStatusMetadata.swift (hunk 1 above)
Comment (andresilveirah) [reply ConsentViewController/Classes/SourcePointClient/ConsentStatusMetadata.swift:16]: It was always hardcoded `false`, but in another point of the code. In this line, I hardcode directly in the class, so other parts of the code don't have to set it to `false`.

File: ConsentViewController/Classes/SourcePointClient/SourcepointClientCoordinator.swift (hunk 2 above)
Comment (andresilveirah) [reply ConsentViewController/Classes/SourcePointClient/SourcepointClientCoordinator.swift:719]: Yes. To give you more context, the SDK stores an object called `gdprMetaData` in the response of `/meta-data`. But we never clear those objects. So, if in the past, the got a response for GDPR, but the config no longer has a GDPR campaign, we would still have a `gdprMetaData` object stored. 

This change basically checks if we have the `gdprMetaData` stored _and_ if gdpr is part of the config. 

Same for CCPA and USNat

**Pull request merged**